dialog/file.go

```diff
@@ -165,7 +165,7 @@ func (f *fileDialog) refreshDir(dir string) {
 		if file.IsDir() {
 			icons = append(icons, f.newFileItem(theme.FolderIcon(), itemPath))
 		} else {
-			icons = append(icons, f.newFileItem(theme.FileIcon(), itemPath))
+			icons = append(icons, f.newFileItem(nil, itemPath))
```

Comment:
Passing nil as information is a bit of an anti-pattern. From what I can tell the icon parameter is now initialising a cache so it probably should be an internal detail.
Instead the folder icon could be handled by the mime code so we don't need this special case.